src/storage.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+const fs = require("fs");
+const fsextra = require("fs-extra");
+const path = require("path");
+const crypto = require("crypto");
+const helper = require("./helper");
+
+class Storage {
+	constructor() {
+		this.references = new Map();
+
+		// Ensures that a directory is empty.
+		// Deletes directory contents if the directory is not empty.
+		// If the directory does not exist, it is created.
+		fsextra.emptyDirSync(helper.getStoragePath());
```

Comment:
So basically I can't implement this in the current state. Images will keep existing if there's any message in memory (buffer) that references it. If we are going to delete images before they are removed from buffer, we will need to find all references and remove them then.

Right now it is not possible to re-download if it's not on the disk.